lib/rules/strict.js

```diff
@@ -82,46 +82,46 @@ module.exports = function(context) {
     }
 
     //--------------------------------------------------------------------------
-    // "never" mode
+    // If this is modules, all "use strict" directives are unnecessary.
     //--------------------------------------------------------------------------
 
-    if (mode === "never") {
+    if (context.parserOptions.sourceType === "module") {
```

Comment:
Hmm, wasn't aware of that. It's true that `context.ecmaFeatures.modules` is the same as `context.parserOptions.sourceType` in that both are just flags to the parser, but those are probably not the best way to check if the parsed code represents a module. I'll update the migration docs. For now, can you make the change in this rule? We can worry about the others later.